### PR TITLE
Use pagination properly to fetch user apps

### PR DESCRIPTION
**Pull request by tivaliy**
It's better to re-use existing method `get_paged_app_instances` to fetch user apps with pagination properly, since current approach allows us to fetch only fixed number of apps (now 100).

### CODE REVIEW

**Comment by tivaliy** [comment]
How to make quick UAT. On CR side we will have the following implementation:

```
In [1]: def get_all_instances(model_name, client_method_name, *args, **kwargs): 
   ...:     client = AppInstanceClient(*client_args) 
   ...:     is_done = False 
   ...:     next_url = None 
   ...:     instances = {} 
   ...:     while not is_done: 
   ...:         paged_result = getattr(client, client_method_name)( 
   ...:             *args, limit=200, url=next_url, **kwargs) 
   ...:         for instance in paged_result.result: 
   ...:             instances[instance.id] = instance 
   ...:         if paged_result.is_last_page(): 
   ...:             is_done = True 
   ...:         else: 
   ...:             next_url = paged_result.next_url 
   ...:     return instances 
   ...:                                                                                                                                                  

In [2]: client_args = ('URL', 'SOME_SETTINGS') 
   ...: from okta import AppInstanceClient    
   ...: apps_client = AppInstanceClient(*client_args)                                                                                                    

In [3]: def get_all_user_apps(user_id, expand_user=True): 
   ...:     expand = f"user/{user_id}" if expand_user else None 
   ...:     filter_string = f"user.id eq \"{user_id}\"" 
   ...:     return get_all_instances('app', 'get_paged_app_instances', 
   ...:                              filter_string=filter_string, expand=expand) 
   ...:                                                                                                                                                  

In [4]: user_id = '00uj9bveacmWTmjM60h7'

In [3]: apps = get_all_user_apps(user_id)                                                                                                                                                            

In [4]: len(apps)                                                                                                                                                                                    
Out[4]: 105
```

While the current implementation is not working properly 

```
In [4]: apps = apps_client.get_app_assignments_for_user(user_id)                                                                                                                                     

In [5]: len(apps)                                                                                                                                                                                    
Out[5]: 100
```

File: okta/AppInstanceClient.py [hunk 1]
```diff
@@ -131,17 +135,6 @@ def deactivate_app_instance(self, id):
 
     # USER
 
-    def get_app_assignments_for_user(self, user, expand_user=True):
```
Comment (drdaeman) [inline okta/AppInstanceClient.py:134]: Why remove `get_app_assignments_for_user` rather than rewriting it as a wrapper over `get_all_instances`?

**Review by drdaeman — COMMENTED** [review]
LGTM, but I'm not sure about removal of an existing method.

File: okta/AppInstanceClient.py (hunk 1 above)
Comment (tivaliy) [reply okta/AppInstanceClient.py:134]: Actually `get_all_instances` doesn't exist in okta-sdk. It's implemented in CR, so this snippet is mostly from CloudRunner. So, I decided to make CR and okta-sdk code consistent and do not create extra methods here. All necessary wrappers will be implemented in CR:

Here is how it looks in CR in `api.utils.app`:

```
def get_all_user_apps(user_id, expand_user=True):
    expand = f"user/{user_id}" if expand_user else None
    filter_string = f"user.id eq \"{user_id}\""
    return utils.get_all_instances('app', 'get_paged_app_instances',
                                   filter_string=filter_string, expand=expand)
```

And here how `get_all_instances` is implemented in `api.utils.base`

```
def get_all_instances(model_name, client_method_name, *args, **kwargs):
    client = get_okta_client(model_name)
    is_done = False
    next_url = None
    instances = {}
    while not is_done:
        paged_result = getattr(client, client_method_name)(
            *args, limit=200, url=next_url, **kwargs)
        for instance in paged_result.result:
            instances[instance.id] = instance
        if paged_result.is_last_page():
            is_done = True
        else:
            next_url = paged_result.next_url
    return instances
```

**Review by yekibud — APPROVED** [review]

File: okta/AppInstanceClient.py (hunk 1 above)
Comment (drdaeman) [reply okta/AppInstanceClient.py:134]: Alright, now I get it. Sorry for the confusion.

**Review by drdaeman — APPROVED** [review]
:shipit:

**Pull request merged**